packages/java-parser/src/index.js

```diff
@@ -2,8 +2,20 @@
 const JavaLexer = require("./lexer");
 const JavaParser = require("./parser");
 
-// const startTime = new Date().getTime();
-const parser = new JavaParser();
+function initializeParser() {
+  let serializedGrammar;
+  try {
+    serializedGrammar = require("./gen/grammar.json");
+  } catch (err) {
+    //
+  }
+  const parser = new JavaParser(serializedGrammar);
+  //generate the grammar.
+  require("../scripts/gen-grammar");
```

Comment:
why do we regenerate the serialized format on parser initialization ?
Would initializing the parser change the grammar?

Reply:
sorry, it seems by refactoring my code i removed a feature, but originally I meant to only generate the grammar if it didn't find an initial grammar.json file

Reply:
Did you re-benchmark after removing this redundant init?

Reply:
I'm not using the same computer but I get NPM 41.35 ops/sec vs Local 45.78 ops/sec